tests/reg_tests/test_adjoint.py

```diff
@@ -87,7 +87,6 @@ def span(val, geo):
         "ref_file": "adjoint_euler_scalar_jst_tut_wing.json",
         "aero_prob": ap_tutorial_wing,
         "evalFuncs": ["cl", "cd"],
-        "N_PROCS": 1,
```

Comment:
@joanibal can you explain what this does? I removed it here because I didn't see it getting used anywhere, but can add it back if it does something.

Reply:
That's for the single core test. It overrides the attributes which specifies the number of cores from the base class used by parametrized. 

If we don't want a single core test than we can get rid of it.

Reply:
Oh I see, those become class attributes when the parameterized class is instantiated? Did you check that this actually works, i.e. testflo calls it with 1 proc? That's pretty neat

Reply:
Exactly, the parameterized_class decorator is actually really simple. 
```
    def decorator(base_class):
        test_class_module = sys.modules[base_class.__module__].__dict__
        for idx, input_dict in enumerate(input_dicts):
            test_class_dict = dict(base_class.__dict__)
            test_class_dict.update(input_dict)

            name = class_name_func(base_class, idx, input_dict)

            test_class_module[name] = type(name, (base_class, ), test_class_dict)
```


It works, I even rechecked before commenting haha

Reply:
Very cool! I recall `paraeterized` saying unittest only supports `@parameterized.expand`, but I guess it works here because we're not actually using unittest, i.e. calling `python test_adjoint.py`, and testflo supports these decorators? Pretty neat stuff, I'll keep this in mind when I parameterize pyoptsparse tests.

Reply:
The tests will run just fine with `python test_adjoint.py	 (I just ran it to check). I'm not entirely sure why though.

The docs for parametrized are kinda confusing about that. They say only @parameterized.exapand will work with unittest but then they show the @paramertized_class working...